src/main/java/com/_up/megastore/services/implementations/UserService.java

```diff
@@ -1,68 +1,79 @@
 package com._up.megastore.services.implementations;
 
 import com._up.megastore.controllers.requests.SignUpRequest;
+import com._up.megastore.data.model.Token;
 import com._up.megastore.data.model.User;
+import com._up.megastore.data.repositories.ITokenRepository;
 import com._up.megastore.data.repositories.IUserRepository;
 import com._up.megastore.services.interfaces.IEmailService;
 import com._up.megastore.services.interfaces.IUserService;
 import com._up.megastore.services.mappers.UserMapper;
+import java.time.LocalDateTime;
 import java.util.UUID;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 
 import java.util.NoSuchElementException;
 
+import static java.util.UUID.randomUUID;
+
 @Service
 public class UserService implements IUserService {
 
   private final IEmailService emailService;
   private final IUserRepository userRepository;
+  private final ITokenRepository tokenRepository;
   private final PasswordEncoder passwordEncoder;
 
   @Value("${frontend.url}")
   private String frontendURL;
 
   public UserService(IUserRepository userRepository, PasswordEncoder passwordEncoder,
-      IEmailService emailService) {
+      IEmailService emailService, ITokenRepository tokenRepository) {
     this.emailService = emailService;
     this.passwordEncoder = passwordEncoder;
     this.userRepository = userRepository;
+    this.tokenRepository = tokenRepository;
   }
 
   @Override
   public void saveUser(SignUpRequest createUserRequest) {
     ifEmailAlreadyExistsThrowException(createUserRequest.email());
     ifUsernameAlreadyExistsThrowException(createUserRequest.username());
     User newUser = createUser(createUserRequest);
-    sendActivationEmail(newUser);
+    UUID activationTokenUUID = generateActivationToken(newUser);
+    sendActivationEmail(newUser,activationTokenUUID);
     userRepository.save(newUser);
   }
 
   @Override
   public User findUserByUsernameOrThrowException(String username) {
     return userRepository.findByUsername(username)
-            .orElseThrow(() -> new NoSuchElementException("User with username "+username+" does not exist."));
+            .orElseThrow(() -> new NoSuchElementException("User with username " + username + " does not exist."));
```

Comment:
Cambiemos el tipo de exception a este

```suggestion
            .orElseThrow(() -> new ResponseStatusException(HttpStatus.BAD_REQUEST, "User with username " + username + " does not exist."));
```